### DISCUSSION TITLE
Diagrams with Mermaid is not rendered

### DISCUSSION THREAD

**Discussion by ayaco-zg** [google/docsy#1692] [Q&A] [answered]
Following [this document](https://www.docsy.dev/docs/adding-content/diagrams-and-formulae/#diagrams-with-mermaid) and #1588, in Hugo v0.119.0, I wrote a Mermaid code block in markdown with no `[params.mermaid]` section in Hugo.config, but it is not rendered as a diagram.

I examined Hugo's output and found the following.
- `<script src="...mermaid.min.js"...>` output by [scripts.html l.81](https://github.com/google/docsy/blob/main/layouts/partials/scripts.html#L81) was correct.
- Mermaid initialization script[^script] that should exist at the end of the main.min....js file output by [scripts.html l.87](https://github.com/google/docsy/blob/main/layouts/partials/scripts.html#L87) was missing.

Please let me know how to address this issue.

[^script]: `function(e) {
    var t, n, o, s = !1;
    if (e(".mermaid").length > 0 && (s = !0),
    !s) {
        mermaid.initialize({
            startOnLoad: !1
        });
        return
    }
    o = {
        enable: !1
    },
    t = function(e, n) {
        var s = {};
        for (const o in e) {
            const i = o.toLowerCase();
            e.hasOwnProperty(o) && n.hasOwnProperty(i) && (typeof e[o] == "object" ? s[o] = t(e[o], n[i]) : s[o] = n[i])
        }
        return s
    }
    ,
    n = t(mermaid.mermaidAPI.defaultConfig, o),
    n.startOnLoad = !0,
    mermaid.initialize(n)
}(jQuery)`

**Comment by ayaco-zg** [comment]
I found another condition.
Occurs when there is `%%{init:{'theme': '.....'}}%%` in Mermaid code block.
For example as follows.
````
```mermaid
%%{init:{'theme': 'neutral'}}%%
flowchart LR
  Start --> End
```
````

Note that if there is at least one Mermaid code block without `%%{init:{'theme': '.....'}}%%` on the same page, as shown below, it was displayed correctly.
````
```mermaid
flowchart LR
  Start --> End
```

```mermaid
%%{init:{'theme': 'neutral'}}%%
flowchart LR
  Start --> End
```
````

Instead of defining `%%{init:{'theme': '.....'}}%%` in Mermaid code block, there is another way to define `[params.mermaid] theme="....."` in hugo.toml.
However, currently I do not want to adopt this method because of the #1588 problem.

Is there any other way to do this?
Also, if this is a bug, is there any way to have it fixed along with issue #1588?

**Comment by curranjohn46** [comment]
I was excited to try mermaid in docsy but it does not seem to work. I cloned the docsy repo and tried the mermaid code but no luck. I am running hugo version:

hugo v0.120.3-a4892a07b41b7b3f1f143140ee4ec0a9a5cf3970+extended

**Reply by curranjohn46** [reply]
I just got mermaid working. Despite have the latest hugo version I still needed to add this to hugo.toml. I added nothing else anywhere.

[params.mermaid]
enable = true

**Comment by hulcher** [comment, marked as answer]
Was struggling with per-diagram themeVariables and found this comment, led me to the right place for figuring out why that was broken.

https://github.com/google/docsy/issues/1882#issue-2161794123

**Reply by ayaco-zg** [reply]
Belatedly, I have confirmed today that this issue has been resolved.
Thanks a lot.
